group-coordinator/src/main/java/org/apache/kafka/coordinator/group/classic/ClassicGroup.java

```diff
@@ -956,6 +956,20 @@ public Optional<OffsetExpirationCondition> offsetExpirationCondition() {
         return Optional.empty();
     }
 
+    /**
+     * The state of this group.
+     *
+     * @return The current state as a lowercase String.
+     */
+    public String stateAsLowerCaseString() {
+        return this.state.toLowerCaseString();
+    }
+
+    @Override
+    public boolean isInStates(Set<String> statesFilter, long committedOffset) {
+        return statesFilter.contains(this.stateAsLowerCaseString()) || statesFilter.contains(this.stateAsString());
```

Comment:
I am not sure to understand why do we need to check both versions here? If the `statesFilter` are lowercased, the first one should be enough, no?

Reply:
In my opinion, the `statefilter` cannot be guaranteed to be in lowercase, there might be that the method `isinstates` is called without doing the lowercase conversion, so needs to check for both versions here.

Reply:
Yeah, it is not but in this context we know that it is. Could we address this by clarifying the requirement in the javadoc? My point is that if `statesFilter` is not lowercased, it won't work anyway because you could also received something like `sTable`.

Reply:
Done.